circleci: failed tests: TestRaceWithBackfill

The following test appears to have failed:

[#19875](https://circleci.com/gh/cockroachdb/cockroach/19875):

```
W160705 19:55:07.177305 kv/dist_sender.go:727  failed to invoke EndTransaction [/Table/51/3/4000/0,/Min): txn aborted "sql/backfill.go:460 (*SchemaChanger).backfillIndexesChunk" id=a85fa0f9 key=/Table/51/3/4000/0 rw=true pri=0.02546962 iso=SERIALIZABLE stat=ABORTED epo=0 ts=1467748507.101656163,0 orig=1467748507.101656163,0 max=1467748507.101656163,0 wto=false rop=false
I160705 19:55:07.177390 internal/client/db.go:474  failed batch: txn aborted "sql/backfill.go:460 (*SchemaChanger).backfillIndexesChunk" id=a85fa0f9 key=/Table/51/3/4000/0 rw=true pri=0.02546962 iso=SERIALIZABLE stat=ABORTED epo=0 ts=1467748507.101656163,0 orig=1467748507.101656163,0 max=1467748507.151656163,0 wto=false rop=false
I160705 19:55:07.347276 kv/txn_coord_sender.go:246  txn coordinator: 28.65 txn/sec, 96.51/43.97/3.49/0.00 %cmmt/cmmt1pc/abrt/abnd, 14ms/70ms/994ms avg/σ/max duration, 0.0/0.2/2 avg/σ/max restarts (247 samples)
I160705 19:55:07.433196 storage/raft_log_queue.go:135  truncating the raft log of range 7 to 234
I160705 19:55:08.390977 rpc/clock_offset.go:135  1 of 1 nodes are within the maximum offset of 50ms
panic: test timed out after 2m0s

goroutine 48746 [running]:
panic(0x15e6640, 0xc8237c6d70)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
testing.startAlarm.func1()
	/usr/local/go/src/testing/testing.go:725 +0x14b
created by time.goFunc
	/usr/local/go/src/time/sleep.go:129 +0x3a

goroutine 1 [chan receive]:
testing.RunTests(0x1e3fce8, 0x2639940, 0x5e, 0x5e, 0xc820277d01)
	/usr/local/go/src/testing/testing.go:583 +0x8d2
testing.(*M).Run(0xc820032f08, 0xc8202568a0)
	/usr/local/go/src/testing/testing.go:515 +0x81
github.com/cockroachdb/cockroach/sql_test.TestMain(0xc820032f08)
	/go/src/github.com/cockroachdb/cockroach/sql/main_test.go:240 +0x70
main.main()
	github.com/cockroachdb/cockroach/sql/_test/_testmain.go:598 +0x114

goroutine 17 [syscall, 2 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1998 +0x1

goroutine 5 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x294eb60)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1011 +0x64
created by github.com/cockroachdb/cockroach/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:598 +0x8a

goroutine 48743 [runnable]:
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).heartbeatLoop(0xc8212fd200, 0x7fa607f46780, 0xc821689fb0, 0x104af41b528551e8, 0x5f05dd62d8b12b81)
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:601 +0x5ec
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).updateState.func2()
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:871 +0x6e
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask.func1(0xc820c34000, 0x234d98d, 0x16, 0x366, 0xc82095c4b0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:184 +0x62
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:185 +0x124

goroutine 47392 [chan receive]:
github.com/cockroachdb/cmux.(*muxListener).Accept(0xc820eefe00, 0x0, 0x0, 0x0, 0x0)
	<autogenerated>:9 +0x125
google.golang.org/grpc.(*Server).Serve(0xc821297830, 0x7fa607f454a0, 0xc820eefe00, 0x0, 0x0)
	/go/src/google.golang.org/grpc/server.go:281 +0x1cf
github.com/cockroachdb/cockroach/server.(*Server).Start.func7()
	/go/src/github.com/cockroachdb/cockroach/server/server.go:365 +0x46
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820eba0c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47391 [chan receive]:
github.com/cockroachdb/cmux.(*muxListener).Accept(0xc820eba020, 0x0, 0x0, 0x0, 0x0)
	<autogenerated>:9 +0x125
crypto/tls.(*listener).Accept(0xc820eba080, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/tls.go:52 +0x60
net/http.(*Server).Serve(0xc8211eca80, 0x7fa607f454d8, 0xc820eba080, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:2117 +0x129
github.com/cockroachdb/cockroach/server.(*Server).Start.func6()
	/go/src/github.com/cockroachdb/cockroach/server/server.go:361 +0x3a
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820eba0a0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47387 [chan receive]:
github.com/cockroachdb/cockroach/server.(*Server).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/server/server.go:317 +0x62
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820eefdc0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47419 [select]:
github.com/cockroachdb/cockroach/server.(*Server).startSampleEnvironment.func1()
	/go/src/github.com/cockroachdb/cockroach/server/server.go:566 +0x15d
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc82163e600)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47397 [select]:
github.com/cockroachdb/cockroach/storage.(*bookie).start.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/reservation.go:255 +0x262
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820eba380)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47372 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:372 +0x31b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820478de0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47443 [select]:
github.com/cockroachdb/cockroach/rpc.(*Context).runHeartbeat(0xc82084bd40, 0xc8203e4a50, 0xc820e530f0, 0xf, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/context.go:263 +0x68d
github.com/cockroachdb/cockroach/rpc.(*Context).GRPCDial.func1.1()
	/go/src/github.com/cockroachdb/cockroach/rpc/context.go:177 +0x66
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820d3dda0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47377 [IO wait]:
net.runtime_pollWait(0x7fa607dfac08, 0x72, 0x0)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc820ea4b50, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820ea4b50, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc820ea4af0, 0x0, 0x7fa607f16060, 0xc8242f8c60)
	/usr/local/go/src/net/fd_unix.go:426 +0x27c
net.(*TCPListener).AcceptTCP(0xc8204b0208, 0xc820f07ed8, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820fb3f00)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47374 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:372 +0x31b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820479200)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47406 [select]:
github.com/cockroachdb/cockroach/storage.(*bookie).start.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/reservation.go:255 +0x262
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820e9fde0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47369 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:372 +0x31b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820478ce0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47421 [select]:
github.com/cockroachdb/cockroach/server.(*Node).startWriteSummaries.func1()
	/go/src/github.com/cockroachdb/cockroach/server/node.go:638 +0x35c
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc82163e620)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47411 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1106 +0x3cf
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc82163e400)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47382 [select]:
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).startStats(0xc8212fd200)
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:203 +0xd21
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).(github.com/cockroachdb/cockroach/kv.startStats)-fm()
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:188 +0x20
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820e529e0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47378 [semacquire]:
sync.runtime_Semacquire(0xc8229c8efc)
	/usr/local/go/src/runtime/sema.go:47 +0x26
sync.(*WaitGroup).Wait(0xc8229c8ef0)
	/usr/local/go/src/sync/waitgroup.go:127 +0xb4
github.com/cockroachdb/cockroach/sql_test.runSchemaChangeWithOperations(0xc8210ba000, 0xc82050b810, 0xc8208dceb0, 0x1b5c7a0, 0x25, 0xfa0, 0x5, 0xc821b93260)
	/go/src/github.com/cockroachdb/cockroach/sql/schema_changer_test.go:485 +0x119c
github.com/cockroachdb/cockroach/sql_test.TestRaceWithBackfill(0xc8210ba000)
	/go/src/github.com/cockroachdb/cockroach/sql/schema_changer_test.go:594 +0xdd4
testing.tRunner(0xc8210ba000, 0x263a060)
	/usr/local/go/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0x892

goroutine 47393 [chan receive]:
github.com/cockroachdb/cmux.(*muxListener).Accept(0xc820eefde0, 0x0, 0x0, 0x0, 0x0)
	<autogenerated>:9 +0x125
github.com/cockroachdb/cockroach/util/netutil.(*Server).ServeWith(0xc8204b0200, 0x7fa607f454a0, 0xc820eefde0, 0xc820706080, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/util/netutil/net.go:111 +0x67
github.com/cockroachdb/cockroach/server.(*Server).Start.func8()
	/go/src/github.com/cockroachdb/cockroach/server/server.go:373 +0x9b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820e90570)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47408 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).processRaft.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:2180 +0xa71
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc82163e340)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47370 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:372 +0x31b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820478d60)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47389 [IO wait]:
net.runtime_pollWait(0x7fa607f44da8, 0x72, 0x0)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc820ea4bc0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820ea4bc0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc820ea4b60, 0x0, 0x7fa607f16060, 0xc8204780c0)
	/usr/local/go/src/net/fd_unix.go:426 +0x27c
net.(*TCPListener).AcceptTCP(0xc8204b0230, 0x5fbe40, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820eba040)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47416 [select]:
github.com/cockroachdb/cockroach/server.(*Node).startComputePeriodicMetrics.func1()
	/go/src/github.com/cockroachdb/cockroach/server/node.go:604 +0x23b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc82163e520)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47417 [select]:
github.com/cockroachdb/cockroach/server.(*Node).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/server/node.go:571 +0x406
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc82163e540)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47385 [chan receive]:
github.com/cockroachdb/cockroach/util/netutil.MakeServer.func2()
	/go/src/github.com/cockroachdb/cockroach/util/netutil/net.go:94 +0x60
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820eefca0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47428 [IO wait]:
net.runtime_pollWait(0x7fa607f44f28, 0x72, 0xc82116a000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc82113a1b0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82113a1b0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82113a150, 0xc82116a000, 0x2000, 0x2000, 0x0, 0x7fa607f36050, 0xc82000e058)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc8201621b0, 0xc82116a000, 0x2000, 0x2000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
google.golang.org/grpc.(*Server).handleRawConn(0xc821297830, 0x7fa607f16720, 0xc821628f20)
	/go/src/google.golang.org/grpc/server.go:318 +0x4ee
created by google.golang.org/grpc.(*Server).Serve
	/go/src/google.golang.org/grpc/server.go:290 +0x38c

goroutine 47423 [select]:
google.golang.org/grpc/transport.(*http2Server).controller(0xc821296ab0)
	/go/src/google.golang.org/grpc/transport/http2_server.go:660 +0x5da
created by google.golang.org/grpc/transport.newHTTP2Server
	/go/src/google.golang.org/grpc/transport/http2_server.go:134 +0x84f

goroutine 47390 [chan receive]:
github.com/cockroachdb/cmux.(*muxListener).Accept(0xc820eba000, 0x0, 0x0, 0x0, 0x0)
	<autogenerated>:9 +0x125
net/http.(*Server).Serve(0xc8211ecb00, 0x7fa607f454a0, 0xc820eba000, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:2117 +0x129
github.com/cockroachdb/cockroach/server.(*Server).Start.func5()
	/go/src/github.com/cockroachdb/cockroach/server/server.go:354 +0x47
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820eba060)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47379 [chan receive]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc8217a82a0)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:299 +0x3a
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:300 +0xada

goroutine 47383 [select]:
github.com/cockroachdb/cockroach/sql.(*LeaseManager).RefreshLeases.func1()
	/go/src/github.com/cockroachdb/cockroach/sql/lease.go:1071 +0xfe8
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820da79b0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47388 [chan receive]:
github.com/cockroachdb/cockroach/server.(*Server).Start.func3()
	/go/src/github.com/cockroachdb/cockroach/server/server.go:338 +0x59
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820eefea0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47422 [select]:
github.com/cockroachdb/cockroach/sql.(*SchemaChangeManager).Start.func1()
	/go/src/github.com/cockroachdb/cockroach/sql/schema_changer.go:640 +0x1762
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc82163e640)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47768 [IO wait]:
net.runtime_pollWait(0x7fa607f44aa8, 0x72, 0xc8204d7800)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc8242f3950, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8242f3950, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8242f38f0, 0xc8204d7800, 0x400, 0x400, 0x0, 0x7fa607f36050, 0xc82000e058)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc8201635b8, 0xc8204d7800, 0x400, 0x400, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
github.com/cockroachdb/cockroach/util/netutil.(*Server).ServeWith.func1(0xc8204b0200, 0x7fa607f16720, 0xc8201bd550, 0xc820706080)
	/go/src/github.com/cockroachdb/cockroach/util/netutil/net.go:131 +0x62
created by github.com/cockroachdb/cockroach/util/netutil.(*Server).ServeWith
	/go/src/github.com/cockroachdb/cockroach/util/netutil/net.go:133 +0x333

goroutine 47380 [chan receive]:
github.com/cockroachdb/cockroach/rpc.NewContext.func1()
	/go/src/github.com/cockroachdb/cockroach/rpc/context.go:109 +0x57
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820eef680)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47410 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).Start.func4()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1035 +0x13d
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc82163e3e0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47424 [IO wait]:
net.runtime_pollWait(0x7fa607dfacc8, 0x72, 0xc8204f6400)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc82113abc0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82113abc0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82113ab60, 0xc8204f6400, 0x400, 0x400, 0x0, 0x7fa607f36050, 0xc82000e058)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc8201624b8, 0xc8204f6400, 0x400, 0x400, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
google.golang.org/grpc/transport.(*http2Client).reader(0xc82069e000)
	/go/src/google.golang.org/grpc/transport/http2_client.go:795 +0x109
created by google.golang.org/grpc/transport.newHTTP2Client
	/go/src/google.golang.org/grpc/transport/http2_client.go:174 +0xd21

goroutine 47394 [chan receive]:
github.com/cockroachdb/cockroach/gossip.(*server).start.func3()
	/go/src/github.com/cockroachdb/cockroach/gossip/server.go:301 +0x5c
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820eba140)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47425 [select]:
google.golang.org/grpc/transport.(*http2Client).controller(0xc82069e000)
	/go/src/google.golang.org/grpc/transport/http2_client.go:873 +0x5da
created by google.golang.org/grpc/transport.newHTTP2Client
	/go/src/google.golang.org/grpc/transport/http2_client.go:201 +0x15c2

goroutine 48745 [runnable]:
github.com/cockroachdb/cockroach/roachpb.(*KeyValue).Unmarshal(0xc82383d8c0, 0xc8202ad007, 0x28, 0xe3b, 0x1, 0xc82383d8c0)
	/go/src/github.com/cockroachdb/cockroach/roachpb/data.pb.go:1695
github.com/cockroachdb/cockroach/config.(*SystemConfig).Unmarshal(0xc824313680, 0xc8202ad005, 0xe3d, 0xe3d, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/config/config.pb.go:531 +0x695
github.com/gogo/protobuf/proto.UnmarshalMerge(0xc8202ad005, 0xe3d, 0xe3d, 0x7fa607efc880, 0xc824313680, 0x0, 0x0)
	/go/src/github.com/gogo/protobuf/proto/decode.go:316 +0xb5
github.com/gogo/protobuf/proto.Unmarshal(0xc8202ad005, 0xe3d, 0xe3d, 0x7fa607efc880, 0xc824313680, 0x0, 0x0)
	/go/src/github.com/gogo/protobuf/proto/decode.go:304 +0x69
github.com/cockroachdb/cockroach/roachpb.Value.GetProto(0xc8202ad000, 0xe42, 0xe42, 0x145e7d7e2a70e207, 0xc800000000, 0x7fa607efc880, 0xc824313680, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/roachpb/data.go:479 +0x2e6
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask.func1(0xc820c34000, 0x2353c52, 0x13, 0x112, 0xc8237c6b60)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:184 +0x62
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:185 +0x124

goroutine 48266 [IO wait]:
net.runtime_pollWait(0x7fa607dfafc8, 0x72, 0xc82094b400)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc820792530, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820792530, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8207924d0, 0xc82094b400, 0x400, 0x400, 0x0, 0x7fa607f36050, 0xc82000e058)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc8204b0e58, 0xc82094b400, 0x400, 0x400, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
github.com/cockroachdb/cockroach/sql_test.runSchemaChangeWithOperations.func1(0xc82050b810, 0x1b5c7a0, 0x25, 0xc8210ba000, 0xc8229c8ef0)
	/go/src/github.com/cockroachdb/cockroach/sql/schema_changer_test.go:425 +0x8c
created by github.com/cockroachdb/cockroach/sql_test.runSchemaChangeWithOperations
	/go/src/github.com/cockroachdb/cockroach/sql/schema_changer_test.go:430 +0x10f

goroutine 47395 [select]:
github.com/cockroachdb/cockroach/gossip.(*Gossip).bootstrap.func1()
	/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:770 +0x237
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820eba160)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47386 [chan receive]:
github.com/cockroachdb/cockroach/util/netutil.MakeServer.func2()
	/go/src/github.com/cockroachdb/cockroach/util/netutil/net.go:94 +0x60
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820eefd20)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47373 [select]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc8211d6e00, 0xecf0e079b, 0x152c17ba, 0x294da80, 0xc820ea0f80, 0xc820c34000, 0xc821036780, 0xc8210c9e78)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:163 +0x5b1
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1.1(0xc821036780, 0xc820db2c60)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:211 +0x78
github.com/cockroachdb/cockroach/storage.(*storeRangeSet).Visit(0xc820db2d20, 0xc824314040)
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:257 +0x246
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:213 +0x21e
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820478e00)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47420 [select]:
github.com/cockroachdb/cockroach/ts.(*poller).start.func1()
	/go/src/github.com/cockroachdb/cockroach/ts/db.go:79 +0x150
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820d5ff90)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47367 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:372 +0x31b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820478c80)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47375 [select]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc8211d71f0, 0xecf0e0796, 0x15319eb1, 0x294da80, 0xc820ea0f80, 0xc820c34000, 0xc821036780, 0xc820ef7e98)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:163 +0x5b1
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1.1(0xc821036780, 0xc820db2c60)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:211 +0x78
github.com/cockroachdb/cockroach/storage.(*storeRangeSet).Visit(0xc820db2ed0, 0xc82040f9c0)
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:257 +0x246
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:213 +0x21e
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820479240)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:142 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:143 +0x62

goroutine 47371 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:372 +0x31b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820c34000, 0xc820478d80
```
Please assign, take a look and update the issue accordingly.

--- a/acceptance/chaos_test.go
+++ b/acceptance/chaos_test.go
@@ -32,6 +32,7 @@ import (
 	"golang.org/x/net/context"
 
 	"github.com/cockroachdb/cockroach/acceptance/cluster"
+	"github.com/cockroachdb/cockroach/testutils"
 	"github.com/cockroachdb/cockroach/util"
 	"github.com/cockroachdb/cockroach/util/log"
 	"github.com/cockroachdb/cockroach/util/randutil"
@@ -146,7 +147,7 @@ func verifyAccounts(t *testing.T, client *testClient) {
 		client.RLock()
 		defer client.RUnlock()
 		err := client.db.QueryRow("SELECT SUM(balance) FROM bank.accounts").Scan(&sum)
-		if err != nil && !isRetryableError(err) {
+		if err != nil && !testutils.IsSQLRetryError(err) {
 			t.Fatal(err)
 		}
 		return err
@@ -163,7 +164,7 @@ func transferMoneyLoop(idx int, state *testState, numAccounts, maxTransfer int) 
 	for !state.done() {
 		if err := transferMoney(client, numAccounts, maxTransfer); err != nil {
 			// Ignore some errors.
-			if !isRetryableError(err) {
+			if !testutils.IsSQLRetryError(err) {
 				// Report the err and terminate.
 				state.errChan <- err
 				break

--- a/acceptance/dynamic_client.go
+++ b/acceptance/dynamic_client.go
@@ -66,13 +66,6 @@ func (dc *dynamicClient) Close() {
 	}
 }
 
-// isRetryableError returns true for any errors that show a connection issue
-// or an issue with the node itself. This can occur when a node is restarting
-// or is unstable in some other way.
-func isRetryableError(err error) bool {
-	return testutils.IsError(err, "(connection reset by peer|connection refused|failed to send RPC|EOF|context deadline exceeded)")
-}
-
 var errTestFinished = errors.New("test is shutting down")
 
 // exec calls exec on a client using a preexisting or new connection.
@@ -82,7 +75,9 @@ func (dc *dynamicClient) exec(query string, args ...interface{}) (gosql.Result, 
 		if err != nil {
 			return nil, err
 		}
-		if result, err := client.Exec(query, args...); err == nil || !isRetryableError(err) {
+		if result, err := client.Exec(
+			query, args...,
+		); err == nil || !testutils.IsSQLRetryError(err) {
 			return result, err
 		}
 	}
@@ -97,7 +92,9 @@ func (dc *dynamicClient) queryRowScan(query string, queryArgs, destArgs []interf
 		if err != nil {
 			return err
 		}
-		if err := client.QueryRow(query, queryArgs...).Scan(destArgs...); err == nil || !isRetryableError(err) {
+		if err := client.QueryRow(
+			query, queryArgs...,
+		).Scan(destArgs...); err == nil || !testutils.IsSQLRetryError(err) {
 			return err
 		}
 	}

--- a/sql/schema_changer_test.go
+++ b/sql/schema_changer_test.go
@@ -430,6 +430,18 @@ ALTER INDEX t.test@foo RENAME TO ufo
 	}
 }
 
+func runSqlWithRetry(t *testing.T, db *gosql.DB, sql string) {
+	for {
+		_, err := db.Exec(sql)
+		if err == nil {
+			break
+		} else if !testutils.IsSQLRetryError(err) {
+			t.Fatal(err)
+		}
+		t.Logf("retry %s on err: %s", sql, err)
+	}
+}
+
 // Run a particular schema change and run some OLTP operations in parallel, as
 // soon as the schema change starts executing its backfill.
 func runSchemaChangeWithOperations(
@@ -473,40 +485,31 @@ func runSchemaChangeWithOperations(
 	for i := 0; i < 10; i++ {
 		k := rand.Intn(maxValue)
 		v := maxValue + i + 1
-		if _, err := sqlDB.Exec(`UPDATE t.test SET v = $2 WHERE k = $1`, k, v); err != nil {
-			t.Fatal(err)
-		}
+		runSqlWithRetry(t, sqlDB, fmt.Sprintf(`UPDATE t.test SET v = %d WHERE k = %d`, v, k))
 		updatedKeys = append(updatedKeys, k)
 	}
 
 	// Reupdate updated values back to what they were before.
 	for _, k := range updatedKeys {
-		if _, err := sqlDB.Exec(`UPDATE t.test SET v = $2 WHERE k = $1`, k, maxValue-k); err != nil {
-			t.Fatal(err)
-		}
+		runSqlWithRetry(t, sqlDB, fmt.Sprintf(`UPDATE t.test SET v = %d WHERE k = %d`, maxValue-k, k))
 	}
 
 	// Delete some rows.
 	deleteStartKey := rand.Intn(maxValue - 10)
 	for i := 0; i < 10; i++ {
-		if _, err := sqlDB.Exec(`DELETE FROM t.test WHERE k = $1`, deleteStartKey+i); err != nil {
-			t.Fatal(err)
-		}
+		runSqlWithRetry(t, sqlDB, fmt.Sprintf(`DELETE FROM t.test WHERE k = %d`, deleteStartKey+i))
 	}
 	// Reinsert deleted rows.
 	for i := 0; i < 10; i++ {
 		k := deleteStartKey + i
-		if _, err := sqlDB.Exec(`INSERT INTO t.test VALUES($1, $2)`, k, maxValue-k); err != nil {
-			t.Fatal(err)
-		}
+		runSqlWithRetry(t, sqlDB, fmt.Sprintf(`INSERT INTO t.test VALUES(%d, %d)`, k, maxValue-k))
 	}
 
 	// Insert some new rows.
 	numInserts := 10
 	for i := 0; i < numInserts; i++ {
-		if _, err := sqlDB.Exec(`INSERT INTO t.test VALUES($1, $2)`, maxValue+i+1, maxValue+i+1); err != nil {
-			t.Fatal(err)
-		}
+		k := maxValue + i + 1
+		runSqlWithRetry(t, sqlDB, fmt.Sprintf(`INSERT INTO t.test VALUES(%d, %d)`, k, k))
 	}
 
 	wg.Wait() // for schema change to complete.
@@ -523,9 +526,7 @@ func runSchemaChangeWithOperations(
 
 	// Delete the rows inserted.
 	for i := 0; i < numInserts; i++ {
-		if _, err := sqlDB.Exec(`DELETE FROM t.test WHERE k = $1`, maxValue+i+1); err != nil {
-			t.Fatal(err)
-		}
+		runSqlWithRetry(t, sqlDB, fmt.Sprintf(`DELETE FROM t.test WHERE k = %d`, maxValue+i+1))
 	}
 }
 
@@ -533,7 +534,6 @@ func runSchemaChangeWithOperations(
 // that run simultaneously.
 func TestRaceWithBackfill(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	t.Skip("#7628")
 
 	var backfillNotification chan bool
 	params, _ := createTestServerParams()
@@ -545,6 +545,7 @@ func TestRaceWithBackfill(t *testing.T) {
 				if backfillNotification != nil {
 					// Close channel to notify that the backfill has started.
 					close(backfillNotification)
+					backfillNotification = nil
 				}
 				return nil
 			},
@@ -669,7 +670,8 @@ CREATE UNIQUE INDEX vidx ON t.test (v);
 	// This test could be made its own test but is placed here to speed up the
 	// testing.
 
-	backfillNotification = make(chan bool)
+	notification := make(chan bool)
+	backfillNotification = notification
 	// Run the schema change in a separate goroutine.
 	var wg sync.WaitGroup
 	wg.Add(1)
@@ -682,7 +684,7 @@ CREATE UNIQUE INDEX vidx ON t.test (v);
 	}()
 
 	// Wait until the schema change backfill starts.
-	<-backfillNotification
+	<-notification
 
 	// Wait for a short bit to ensure that the backfill has likely progressed
 	// and written some data, but not long enough that the backfill has

--- a/testutils/error.go
+++ b/testutils/error.go
@@ -51,6 +51,13 @@ func IsPError(pErr *roachpb.Error, re string) bool {
 	return matched
 }
 
+// IsSQLRetryError returns true if err is retryable. This is true for errors
+// that show a connection issue or an issue with the node itself. This can
+// occur when a node is restarting or is unstable in some other way.
+func IsSQLRetryError(err error) bool {
+	return IsError(err, "(connection reset by peer|connection refused|failed to send RPC|EOF|context deadline exceeded)")
+}
+
 // Caller returns filename and line number info for the specified stack
 // depths. The info is formated as <file>:<line> and each entry is separated
 // for a space.

